torch/csrc/cuda/shared/nvtx.cpp

```diff
@@ -1,22 +1,206 @@
 #ifdef _WIN32
 #include <wchar.h> // _wgetenv for nvtx
 #endif
+
 #include <nvToolsExt.h>
 #include <torch/csrc/utils/pybind.h>
 
 namespace torch {
 namespace cuda {
 namespace shared {
 
+#ifndef USE_ROCM
+
+// registry idea from NVIDIA/nvtx-plugins  @jdekhtiar
+class DomainCategoryRegistry {
+ public:
+  DomainCategoryRegistry() {}
+
+  ~DomainCategoryRegistry() {
+    for (auto& domain : domains)
+      nvtxDomainDestroy(domain.second);
+  }
+
+  nvtxDomainHandle_t getDomain(const std::string& domain_name) {
+    auto it = domains.find(domain_name);
+    if (it != domains.end())
+      return it->second;
```

Comment:
PyTorch is C++17 project now, so let's use some of its functions
Also, why not const auto? You don't plan to modify it, do you?
```suggestion
    if ( const auto it = domains.find(domain_name); it != domains.end()) {
      return it->second;
   }   
```

Reply:
Right, I use more consts in 30804cc